### PR TITLE
DR-1000 Autocomplete lag in filter panels

### PR DESCRIPTION
**Pull request by myessail**

### CODE REVIEW

File: src/modules/bigquery.js [hunk 1]
```diff
@@ -159,6 +160,40 @@ export default class BigQuery {
...
+    dataset,
+    tableName,
+    token,
+    filterStatement,
+    joinStatement,
+  ) => {
+    const url = `https://bigquery.googleapis.com/bigquery/v2/projects/${dataset.dataProject}/queries`;
+    const filterOrEmpty = _.isEmpty(filterStatement)
+      ? `WHERE ${columnName} LIKE '%${currText}%'`
+      : `WHERE ${filterStatement} AND ${columnName} LIKE '%${currText}%'`;
```
Comment (fboulnois) [inline src/modules/bigquery.js]: Will this kick off for every character entered in the filter panel?

**Review by fboulnois — COMMENTED** [review]
One question:

File: src/modules/bigquery.js (hunk 1 above)
Comment (myessail) [reply src/modules/bigquery.js]: nope, i'm calling `getAutocompleteForColumnDebounced` in the component which will fire this off after the user stops typing for 500ms

Comment (fboulnois) [reply src/modules/bigquery.js]: Nice, that's awesome.

**Review by fboulnois — APPROVED** [review]
lgtm

File: src/modules/bigquery.js [hunk 2]
```diff
@@ -159,6 +160,40 @@ export default class BigQuery {
...
+    const joinOrEmpty = _.isEmpty(joinStatement) ? '' : joinStatement;
```
Comment (mackenziekerwin) [inline src/modules/bigquery.js]: Is this just equal to `joinStatement`? I believe it is a string so in both cases `joinOrEmpty` and `joinStatement` are equal

File: src/modules/bigquery.js (hunk 1 above)
Comment (mackenziekerwin) [inline src/modules/bigquery.js]: I think when `filterStatement` is not empty it already starts with "WHERE". Would this cause a parsing error for BigQuery?

**Review by mackenziekerwin — APPROVED** [review]
Nice job, I just have some quick comments:

File: src/modules/bigquery.js (hunk 2 above)
Comment (myessail) [reply src/modules/bigquery.js]: not quite, if there is no `joinStatement` (it hasn't been created in the store), it'll get put in as `undefined`

Comment (mackenziekerwin) [reply src/modules/bigquery.js]: Ok, got it!!

**Pull request merged**